lib/orocos/process.rb

```diff
@@ -1114,19 +1139,20 @@ def spawn(log_level: nil, working_directory: Orocos.default_working_directory,
                 end
 
                 read.close
-                write.fcntl(Fcntl::F_SETFD, 1)
                 ::Process.setpgrp
                 begin
                     if working_directory
                         Dir.chdir(working_directory)
                     end
-                    exec(*cmdline)
+                    exec(*cmdline, ior_write_fd => ior_write_fd)
```

Comment:
[Kernel#exec](http://www.ruby-doc.org/core/Kernel.html#method-i-exec)

So when the current process is replaced with the process invoked, it does migrate all the file descriptors to the invoked process?

Reply:
and when the invoked returns, it fallbacks to the beginning process (aka ruby script)?

Reply:
Whatever is inside the fork block is what is ran by the forked process. `Kernel#exec` nowadays only pass the file descriptors that were explicitly passed (like I did in this line), the others are closed.